### PR TITLE
:book: Add doc on how to develop CSPO

### PR DESCRIPTION
**Pull request by michal-gubricky**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: docs/develop.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+![tilt](./pics/tilt.png "Tilt")
+
+Operator development requires a lot of iteration, and the “build, tag, push, update deployment” workflow can be very tedious. Tilt makes this process much simpler by watching for updates and automatically building and deploying them. To build a kind cluster and to start Tilt, run:
+
+```shell
+make tilt-up
+```
+
+> To access the Tilt UI please go to: `http://localhost:10350`
```
Comment (chess-knight) [inline docs/develop.md:25]:
```suggestion
> To access the Tilt UI please go to: `http://localhost:10351`
```

File: docs/develop.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+You should make sure that everything in the UI looks green. If not, you can trigger the Tilt workflow again. To establish a connection to your OpenStack project, you must supply a secret containing the `clouds.yaml` file.
+
+```bash
+kubectl create secret generic <my-secret> --from-file=clouds.yaml=path/to/clouds.yaml
+```
+
+To transfer the credentials stored in the secret mentioned above to the operator, the user must create an `OpenStackClusterStackReleaseTemplate` object and specify this secret in the `identityRef` field. Refer to the `examples/cspotemplate.yaml` file for more details. Afterward, apply this template to the local Kind cluster, which was built by the previous `make` command.
```
Comment (chess-knight) [inline docs/develop.md:33]: You can mention also `cloudName` field

File: examples/clusterstack.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  provider: openstack
+  name: ferrol
+  kubernetesVersion: "1.27"
+  channel: stable
+  autoSubscribe: false
+  providerRef:
+    apiVersion: infrastructure.clusterstack.x-k8s.io/v1alpha1
+    kind: OpenStackClusterStackReleaseTemplate
+    name: cspotemplate
+    namespace: default
```
Comment (chess-knight) [inline examples/clusterstack.yaml:16]:
```suggestion
```

File: examples/cspotemplate.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: infrastructure.clusterstack.x-k8s.io/v1alpha1
+kind: OpenStackClusterStackReleaseTemplate
+metadata:
+  name: cspotemplate
```
Comment (chess-knight) [inline examples/cspotemplate.yaml:4]:
```suggestion
  name: cspotemplate
  namespace: cluster
```

File: docs/develop.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+```shell
+make tilt-up
+```
+
+> To access the Tilt UI please go to: `http://localhost:10350`
+
+You should make sure that everything in the UI looks green. If not, you can trigger the Tilt workflow again. To establish a connection to your OpenStack project, you must supply a secret containing the `clouds.yaml` file.
+
+```bash
+kubectl create secret generic <my-secret> --from-file=clouds.yaml=path/to/clouds.yaml
```
Comment (chess-knight) [inline docs/develop.md:30]:
```suggestion
kubectl create secret generic <my-secret> --from-file=clouds.yaml=path/to/clouds.yaml -n cluster
```

**Review by matofeder — APPROVED** [review]
lgtm

**Pull request merged**